### ISSUE TITLE
Using @Named doesn't generate the type parameter

### ISSUE THREAD

**Issue by ResoDev** [Milad-Akarie/injectable#20]
Hello again!
I really enjoy using this great library but because of my strict lint rules (no implicit casts from `dynamic`), I cannot use named dependencies.

My injectable module uses a name for a `HiveBox` because I want to have multiple different instances:
```dart
@registerModule
abstract class InfrastructureInjectableModule {
  // other dependencies here...
  @lazySingleton
  @Named(HiveBox.weatherForecast)
  Future<Box> get weatherForecastBox => Hive.openBox(HiveBox.weatherForecast);
}

abstract class HiveBox {
  static const weatherForecast = 'weatherForecast';
}
```

Then, in a class the constructor looks like this:
```dart
@lazySingleton
@RegisterAs(LocalForecastCache)
class HiveForecastCache implements LocalForecastCache {
  final Box weatherForecastBox;

  HiveForecastCache(@Named(HiveBox.weatherForecast) this.weatherForecastBox);
  ...
}
```

Unfortunately, **injectable** generates a get_it call without the generic parameter. In my case, this results in a `The argument type 'dynamic' can't be assigned to the parameter type 'Box'.dart(argument_type_not_assignable)`

```dart
  g.registerLazySingleton<LocalForecastCache>(() => HiveForecastCache(
        g('weatherForecast'),
      ));
```

Should be `g<Box>('weatherForecast');`

**Comment by Milad-Akarie** [comment]
Hey, @Matej I believe that GetIt will completely ignore the type if you provide an instanceName. Everything works fine if you manually add the type?

**Comment by ResoDev** [comment]
I'm sure that get_it ignores the type but that's not the issue which I'm facing. Although the type returned by get_it will be correct, the Dart static analysis is giving me a hard time because of my strict lint rules.

Basically, although the type parameter is kind of irrelevant for get_it, it's important for static analysis.

**Comment by Milad-Akarie** [comment]
Fixed in injectable_generator: 0.2.4.
